src/explorer/siteNodeBase.ts

```diff
@@ -156,37 +156,28 @@ export class SiteNodeBase extends NodeBase {
             await window.showErrorMessage(`You have not yet opened a folder to deploy.`);
             throw new Error('No open workspace');
         }
+        let publishCredentials;
+        let config: WebSiteModels.SiteConfigResource;
+        let oldDeployment;
 
-        const publishCredentials = !this._isSlot ?
-            await this.webSiteClient.webApps.listPublishingCredentials(this.site.resourceGroup, this._siteName) :
-            await this.webSiteClient.webApps.listPublishingCredentialsSlot(this.site.resourceGroup, this._siteName, this._slotName);
-        const config = !this._isSlot ?
-            await this.webSiteClient.webApps.getConfiguration(this.site.resourceGroup, this._siteName) :
-            await this.webSiteClient.webApps.getConfigurationSlot(this.site.resourceGroup, this._siteName, this._slotName);
-        const oldDeployment = !this._isSlot ?
-            await this.webSiteClient.webApps.listDeployments(this.site.resourceGroup, this._siteName) :
-            await this.webSiteClient.webApps.listDeploymentsSlot(this.site.resourceGroup, this._siteName, this._slotName);
+        if (!this._isSlot) {
+            publishCredentials = await this.webSiteClient.webApps.listPublishingCredentials(this.site.resourceGroup, this._siteName);
+            config = await this.webSiteClient.webApps.getConfiguration(this.site.resourceGroup, this._siteName);
+            oldDeployment = await this.webSiteClient.webApps.listDeployments(this.site.resourceGroup, this._siteName);
+        } else {
+            publishCredentials = await this.webSiteClient.webApps.listPublishingCredentialsSlot(this.site.resourceGroup, this._siteName, util.extractDeploymentSlotName(this.site));
+            config = await this.webSiteClient.webApps.getConfigurationSlot(this.site.resourceGroup, this._siteName, util.extractDeploymentSlotName(this.site));
+            oldDeployment = await this.webSiteClient.webApps.listDeploymentsSlot(this.site.resourceGroup, this._siteName, util.extractDeploymentSlotName(this.site));
+        }
 
         if (config.scmType !== 'LocalGit') {
-            let input;
-            let oldScmType = config.scmType;
-            let updateConfig = config;
-            updateConfig.scmType = 'LocalGit';
-
-            if (oldScmType !== 'None') {
-                input = await window.showWarningMessage(`Deployment source for "${this._siteName}" is set to "${oldScmType}". Change to "LocalGit"?`, 'Yes');
-            }
-
-            if (oldScmType === 'None' || input === 'Yes') {
-                !this._isSlot ?
-                    await this.webSiteClient.webApps.updateConfiguration(this.site.resourceGroup, this._siteName, updateConfig) :
-                    await this.webSiteClient.webApps.updateConfigurationSlot(this.site.resourceGroup, this._siteName, updateConfig, this._slotName);
-            } else {
+            let update = await this.updateScmType(this, config);
+            if (!update) {
```

Comment:
In this case it's hard to tell what the boolean means. For example, I initially thought 'true' meant the update succeeded and 'false' meant it failed. 

Can we move `throw new UserCancelledError()` back into `updateScmType` instead of returning false?